PubSub/src/Subscription.php

```diff
@@ -601,17 +641,31 @@ public function update(array $subscription, array $options = [])
             }
         }
 
-        $subscription = $this->formatSubscriptionDurations($subscription);
+        $maskPaths = [];
+        foreach ($updateMaskPaths as $path) {
+            $maskPaths[] = $this->requestHandler->getSerializer()::toSnakeCase($path);
+        }
 
-        $subscription = [
-            'name' => $this->name
-        ] + $this->formatDeadLetterPolicyForApi($subscription);
+        $fieldMask = new FieldMask([
+            'paths' => $maskPaths
+        ]);
 
-        return $this->info = $this->connection->updateSubscription([
-            'name' => $this->name,
-            'subscription' => $subscription,
-            'updateMask' => implode(',', $updateMaskPaths)
-        ] + $options);
+        $subscription = $this->formatSubscriptionDurations($subscription);
+        $subscription = $this->formatDeadLetterPolicyForApi($subscription);
+
+        $subscriptionProto = $this->requestHandler->getSerializer()->decodeMessage(
+            new SubscriptionProto,
+            [
+                'name' => $this->name
+            ] + $subscription
+        );
+
+        return $this->info = $this->requestHandler->sendRequest(
+            SubscriberClient::class,
+            'updateSubscription',
+            [$subscriptionProto, $fieldMask],
+            $options
+        );
```

Comment:
Honest question but - do we really feel like this is an improvement from what we had? It seems like we are moving the logic around instead of removing it.

Reply:
Here are a few points because of which I feel this is better:
- Handwritten logic is only in the resource class or the caller(_Subscription.php in this case_), previously it was in `Subscription.php` and then `Rest.php` and `Grpc.php`. So, the handwritten surface is minimized.
- The call structure matches what we have in the GAPICs, if the method has 2 required args, then we pass an array of 2 args as the 3rd parameter.
- This looks similar to the connection class calling a method in the current approach, but a big distinguishing factor is that the request handler is just one class, while every product has their own connection classes, which we are removing.
- Although we are still in early phases of that discussion, but this might lead the way to reducing the handwritten layer even more given that we are able to see patterns which are closely related to the GAPIC methods being called.

Reply:
> So, the handwritten surface is minimized.

I can see how having to make changes in 1 place instead of 3 is an advantage

> The call structure matches what we have in the GAPICs

I am confused why this is important, or how it differs from the previous approach

> request handler is just one class, while every product has their own connection classes

 Fewer classes is not a _good_ thing. Where before, the adapter logic had two dedicated classes to contain it (`Rest` and `Grpc`), now the logic is found in each resource classes, which makes it harder to track down.

> this might lead the way to reducing the handwritten layer even more given that we are able to see patterns which are closely related to the GAPIC methods being called.

I don't really understand this